great_expectations/experimental/datasources/config.py

```diff
@@ -2,32 +2,50 @@
 from __future__ import annotations
 
 import logging
+import pathlib
 from pprint import pformat as pf
-from typing import Dict, Type
+from typing import TYPE_CHECKING, Dict, List, Type, Union
 
-from pydantic import Extra, Field, validator
+from pydantic import Extra, Field, ValidationError, validator
+from typing_extensions import Final
 
 from great_expectations.experimental.datasources.experimental_base_model import (
     ExperimentalBaseModel,
 )
 from great_expectations.experimental.datasources.interfaces import Datasource
 from great_expectations.experimental.datasources.sources import _SourceFactories
 
+if TYPE_CHECKING:
+    from pydantic.error_wrappers import ErrorDict as PydanticErrorDict
+
+
 LOGGER = logging.getLogger(__name__)
 
 
-_ZEP_STYLE_DESCRIPTION = "ZEP Experimental Datasources"
+_ZEP_STYLE_DESCRIPTION: Final[str] = "ZEP Experimental Datasources"
+
+_MISSING_XDATASOURCES_ERRORS: Final[List[PydanticErrorDict]] = [
+    {
+        "loc": ("xdatasources",),
+        "msg": "field required",
+        "type": "value_error.missing",
+    }
+]
 
 
 class GxConfig(ExperimentalBaseModel):
     """Represents the full new-style/experimental configuration file."""
 
-    datasources: Dict[str, Datasource] = Field(..., description=_ZEP_STYLE_DESCRIPTION)
+    xdatasources: Dict[str, Datasource] = Field(..., description=_ZEP_STYLE_DESCRIPTION)
```

Comment:
@billdirks 
Somehow I missed that `datasources` IS a top-level key for the old config format. 😅